### DISCUSSION TITLE
cntext2.x | 新增读取pdf/docx| 提取MD&A | 文本可视化等功能

### DISCUSSION THREAD

**Discussion by giscus** [hiDaDeng/hidadeng.github.io#40] [交流]
# cntext2.x | 新增读取pdf/docx| 提取MD&A | 文本可视化等功能

一、cntext 1.1 新增函数 cntext2.1.2新增函数有
 get_cntext_path() 查看cntext2.x的安装路径 read_pdf()/read_docx() 读取 pdf、docx文件 extract_mda() 提取中文年报文本中的管理层讨论与分析 matplotlib_chinese() 支持matplotlib显示中文 lexical_dispersion_plot1() 词汇分散图 lexical_dispersion_plot2() 词汇分散图  已购买cntext2.x的用户，可私信找到大邓获取最新版本安装包！

1.2 安装 所有 cntext2.x 安装方法类似， 以目前 cntext2.1.2 为例，将 cntext-2.1.2-py3-none-any.whl 放置于桌面，打开 cmd (苹果电脑打开terminal)， 输入cd desktop
cd desktop 之后在 cmd (苹果电脑打开terminal) 中使用 pip3 安装
pip3 install distinctiveness pip3 install cntext-2.1.2-py3-none-any.whl 文章开头和文章末都有*** cntext-2.1.2-py3-none-any.whl*** 获取方式说明。
二、实验 2.1 get_cntext_path() 如果你熟悉PYTHON，想对cntext内进行修改， 可以使用该函数找到cntext安装路径。
import cntext as ct ct.get_cntext_path() Run
/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/cntext  不同电脑返回的位置是不同的，以上路径是大邓Mac中cntext2.x的安装路径
 2.2 read_docx() 读取 docx文件。 自己diy一个 test.

https://textdata.cn/blog/2024-05-14-add-readpdf-readdocx-lexical-dispersion-plot/

**Comment by hiDaDeng** [comment]
cntext2新增的都是很赞的功能，重点介绍extract_mda, 这可能是目前全网唯一提取管理层讨论与分析md&a的功能函数。 封装后， 只需两行代码即可提取md&a内容。


加大邓 WeChat: 372335839， 备注「姓名-学校-专业」， 100元领取 cntext-2.1.2-py3-none-any.whl 文件。本文出现的cntext，默认均为2.x版本。
